src/libYARP_dev/src/devices/msgs/ros/include/TickDuration.h

```diff
@@ -30,6 +30,39 @@ class TickDuration : public yarp::os::idl::WirePortable
     {
     }
 
+    TickDuration(double timestamp) :
+            sec(0),
+            nsec(0)
+    {
+        uint64_t time = (uint64_t) (timestamp * 1000000000UL);
+        sec = (time / 1000000000UL);
+        nsec = (time % 1000000000UL);
+        if (sec > UINT_MAX) {
+            yWarning("Timestamp exceeded the 64 bit representation, resetting it to 0");
+            sec = 0;
+        }
+    }
+
+    TickDuration& operator=(const double timestamp)
+    {
+        uint64_t time = (uint64_t) (timestamp * 1000000000UL);
+        sec = (time / 1000000000UL);
+        nsec = (time % 1000000000UL);
+        if (sec > UINT_MAX) {
+            yWarning("Timestamp exceeded the 64 bit representation, resetting it to 0");
+            sec = 0;
+        }
+        return *this;
+    }
+
+    operator double()
+    {
+        if (nsec > 1000000000UL) {
+            yWarning("Check on nsec > 1000000000UL failed");
```

Comment:
We could add  `TickDuration::double(): ` to this print, so it is easier to debug its origin if it pops out of nowhere.

Reply:
fixed